tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -693,7 +693,15 @@ def ensure_runner_stopped(self):
             # This might leak a file handle from the queue
             self.logger.warning("Forcibly terminating runner process")
             self.test_runner_proc.terminate()
-            self.test_runner_proc.join(10)
+
+            # Multiprocessing queues are backed by operating system pipes. If
+            # the pipe in the child process had buffered data at the time of
+            # forced termination, the queue is no longer in a usable state
+            # (subsequent attempts to retrieve items may block indefinitely).
+            # Discard the potentially-corrupted queue and create a new one.
+            self.teardown()
```

Comment:
Calling teardown here seems wrong since we aren't actually tearing down the current object. I think you should just copy in the calls to `close` and then replace the queues with new ones.

Reply:
You got it